go-controller/pkg/mock/mock_ovn.go

```diff
@@ -0,0 +1,469 @@
+package mockovn
+
+import (
+	"fmt"
+	"runtime"
+
+	goovn "github.com/ebay/go-ovn"
+	util "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	aggErrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+)
+
+const (
+	LogicalSwitchType     string = "Logical_Switch"
+	LogicalSwitchPortType string = "Logical_Switch_Port"
+)
+
+const (
+	OpAdd    string = "Add"
+	OpDelete string = "Delete"
+	OpUpdate string = "Update"
+)
+
+// used to update fields for existing objects in mock cache
+type UpdateCache struct {
+	FieldType  string
+	FieldValue interface{}
+}
+
+// object cache for mock ovn client
+type MockObjectCacheByName map[string]interface{}
+
+// provides Execute() interface
+type MockExecution interface {
+	Execute(cmds ...util.OVNCommandInterface) error
+}
+
+// mock ovn client for testing
+type MockOVNClient struct {
+	db string
+	// cache holds ovn db rows by table name as the key
+	cache map[string]MockObjectCacheByName
+	// error injection
+	// keys are of the form: Table:Name:FieldType
+	errorCache map[string]error
+}
+
+var _ util.OVNInterface = &MockOVNClient{}
+
+type MockOVNCommand struct {
+	Exe       MockExecution
+	op        string
+	table     string
+	objName   string
+	obj       interface{}
+	objUpdate UpdateCache
+}
+
+// MockOVNCommand implements OVNCommandInterface
+var _ util.OVNCommandInterface = &MockOVNCommand{}
+
+// return a new mock client to operate on db
+func NewMockOVNClient(db string) *MockOVNClient {
+	return &MockOVNClient{
+		db:         db,
+		cache:      make(map[string]MockObjectCacheByName),
+		errorCache: make(map[string]error),
+	}
+}
+
+func functionName() string {
+	pc, _, _, ok := runtime.Caller(1)
+	if !ok {
+		return "???"
+	}
+
+	fn := runtime.FuncForPC(pc)
+	return fn.Name()
+}
+
+// Client Interface Methods
+
+// TODO: implement mock methods as we keep adding unit-tests
```

Comment:
nit: can you move the not-implemented functions to another file? Makes these things a lot easier to read.

ovn-k has been bad with file size, we should start doing better :-)

Reply:
This file is the one where the non-implemented and common functions of the interface are. As folks implement more and more of the interface, I envision them doing so in individual files. For instance when we add things for LSR, we can have a new file mock_lsr.go and move the relevant API methods there.

Reply:
agreed; just that right now, `mock_ovn.go` has both implementation and a lot of placeholder; could you move the placeholder to another file?

Reply:
done.